execinquery.go

```diff
@@ -45,28 +57,35 @@ func run(pass *analysis.Pass) (interface{}, error) {
 			switch arg := n.Args[i].(type) {
```

Comment:
Maybe also adding this checking before `var s string`?
```go
			if i >= len(n.Args) {
				return
			}
```
I know this is impossible for the current `database/sql`, but I feel we should still check.

Reply:
In the golang, the definition of the variable is different between `var s string` and `s := "AAAA"`
this code is only checking `s := "AAAA"` case.

Reply:
No I was not talking about the variable `s`. I was proposing the following code, in addition to what @idez what already did:

```go
			// ...

			if i >= len(n.Args) {
				return
			}

			var s string
			switch arg := n.Args[i].(type) {

			// ...
```
The reason is that `n.Args[i]` will panic if `i >= en(n.Args[i])`. And the reason I propose putting the check before the line `var s string` is because the variable `s` is meaningful only later.

Reply:
the check is at the beginning of the function.

Reply:
Ah, sorry I somehow missed it.

Reply:
Wait, no, the check is still wrong...

Reply:
FYI: my login is ldez not idez

I will take a look